### PR TITLE
add era5 cloud data

### PR DESCRIPTION
**Pull request by szy21**
Checklist:
- [x] I created a new folder `$artifact_name`
  - [x] I added a `README.md` in that that folder that
    - [x] describes the data and processing done to it
    - [x] lists the sources of the raw data
    - [x] lists the required citation, licenses
  - [x] If applicable (e.g., for Creative Commons), I added a `LICENSE` file
  - [x] I added the scripts that retrieve, process, and produce the artifact
  - [x] I added the environment used for such scripts (typically, `Project.toml`
        and `Manifest.toml`)
  - [x] I added the `OutputArtifacts.toml` file containing the information
        needed for package developers to add `$artifact_name` to their package
- [x] I uploaded the artifact folder to the Caltech cluster (in
      `/groups/esm/ClimaArtifacts/artifacts/$artifact_name`)
- [x] I added the relevant code to the `Overides.toml` on the Caltech Cluster
      (in `/groups/esm/ClimaArtifacts/artifacts/Overrides.toml`)
- [x] I added a link to the main `README.md` to point to the new artifact

### CODE REVIEW

File: era5_cloud/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Cloud data, monthly means, 2010
+
+This artifact repackages data coming from [ERA5 reanalysis](https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels-monthly-means?tab=overview) 
+and contains monthly averaged cloud properties in 2010.
+
+The input file is defined in pressure coordinates. We convert this into altitude over mean-sea level
+using $P = P^*exp(-z / H)$ with scale height $H$. We assume $P^* = 1e5$ (Pa) and $H = 7000$ (m). The output 
+is a NetCDF file that contains cloud fraction, cloud liquid water content (kg/kg), and cloud ice water content (kg/kg) 
+defined on a lon-lat-z-time grid.
```
Comment (Sbozzolo) [inline era5_cloud/README.md:9]: I think we should add support for reading directly pressure levels at some point.

File: era5_cloud/create_artifact.jl [hunk 2]
```diff
@@ -0,0 +1,86 @@
+using Downloads
+using NCDatasets
+
+using ClimaArtifactsHelper
+
+FILE_URL = "https://caltech.box.com/shared/static/929dvfp42shtxwze00vfkk2i2clz79ty.nc"
```
Comment (Sbozzolo) [inline era5_cloud/create_artifact.jl:6]: Would it be possible to have an automated download link from some official source?

Comment (szy21) [reply era5_cloud/create_artifact.jl:6]: That would be more convenient. I downloaded the data here by selecting a bunch of things: https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels-monthly-means?tab=download. They provide an API but it requires a personal access token.

Comment (szy21) [reply era5_cloud/create_artifact.jl:6]: I think it would be helpful if we could set it up. We will download a lot of data from era5 for e.g. calibration.

Comment (Sbozzolo) [reply era5_cloud/create_artifact.jl:6]: We can do this. We can add a token to ClimaCommon enviroments.

File: era5_cloud/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+## Usage
+To recreate the artifact:
+1. Create a python virtual environment
+2. Activate the new virtual env
+3. In the same terminal run `pip install -r requirements.txt`
```
Comment (Sbozzolo) [inline era5_cloud/README.md:15]: Can you add the `reuirements.txt` file?

File: era5_cloud/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+4. In the same terminal run `julia --project create_artifact.jl`
+
+## Requirements
+- Python >= 3
+
+## Reference
+Hersbach, H., Bell, B., Berrisford, P., Biavati, G., Horányi, A., Muñoz Sabater, J., Nicolas, J., Peubey, C., Radu, R., Rozum, I., Schepers, D., Simmons, A., Soci, C., Dee, D., Thépaut, J-N. (2023): ERA5 monthly averaged data on pressure levels from 1940 to present. Copernicus Climate Change Service (C3S) Climate Data Store (CDS), DOI: 10.24381/cds.6860a573 (Accessed on 08-Oct-2024)
```
Comment (Sbozzolo) [inline era5_cloud/README.md:22]: I think there's extra licenses with ERA5 that we need to add here.

File: era5_cloud/download_era5.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import cdsapi
```
Comment (Sbozzolo) [inline era5_cloud/download_era5.py:1]: Can you add instructions/links to set this up?

File: era5_cloud/README.md (hunk 4 above)
Comment (szy21) [reply era5_cloud/README.md:22]: The license information is here (in the right panel): https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels-monthly-means?tab=download. What should we include here?

File: era5_cloud/README.md (hunk 3 above)
Comment (szy21) [reply era5_cloud/README.md:15]: Yes, I forgot to add it.

File: era5_cloud/README.md (hunk 4 above)
Comment (Sbozzolo) [reply era5_cloud/README.md:22]: We should at least include a link to the original license

Comment (szy21) [reply era5_cloud/README.md:23]: I added a link to the licence pdf

File: era5_cloud/download_era5.py (hunk 5 above)
Comment (szy21) [reply era5_cloud/download_era5.py:1]: I added the instruction in the README

**Review by Sbozzolo — APPROVED** [review]

**Pull request merged**